java/ql/src/Telemetry/ExternalAPI.qll

```diff
@@ -0,0 +1,95 @@
+/** Provides classes and predicates related to handling APIs from external libraries. */
+
+private import java
+private import semmle.code.java.dataflow.DataFlow
+private import semmle.code.java.dataflow.DataFlow
+private import semmle.code.java.dataflow.ExternalFlow
+private import semmle.code.java.dataflow.FlowSources
+private import semmle.code.java.dataflow.FlowSummary
+private import semmle.code.java.dataflow.internal.DataFlowPrivate
+private import semmle.code.java.dataflow.TaintTracking
+
+/**
+ * An external API from either the Java Standard Library or a 3rd party library.
+ */
+class ExternalAPI extends Callable {
+  ExternalAPI() { not this.fromSource() }
+
+  /** Holds if this API is a candidate worth supporting */
+  predicate isWorthSupporting() { not isTestLibrary() and not isParameterlessConstructor() }
```

Comment:
Expressing this in a positive sense seems misleading as we are merely excluding a (likely non-exhaustive) list of things that aren't worth supporting.

Reply:
Maybe
```suggestion
  /** Holds if this API is a candidate worth supporting */
  predicate isUninteresting() { isTestLibrary() or isParameterlessConstructor() }
```

Reply:
Alternatively, we could exclude these cases in the charpred.

Reply:
I tried the positive route as it avoided negating the predicate in every query we use it. But agreed, it's a non-exhaustive list. Went with `isWorthSupporting` for now